File: Daily Programmer/Easy Challenges/0286 - Reverse Factorial/Solution.py
def reverse_factorial(n, divisor):
    if n / divisor == 1:
        return n
    return 0 if n % divisor > 0 else reverse_factorial(n / divisor, divisor + 1)


def format_factorial(n):
    return "None" if n == 0 else "%d!" % n


def main():
    num = int(input("Enter an integer: "))
    print("%d = %s" % (num, format_factorial(reverse_factorial(num, 2))))


if __name__ == '__main__':
    main()

File: Daily Programmer/Easy Challenges/0286 - Reverse Factorial/Solution.java
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.print("Enter an integer: ");
        int num = reader.nextInt();
        System.out.printf("%d = %s", num, formatFactorial(reverseFactorial(num, 2)));
    }

    public static int reverseFactorial(int n, int divisor) {
        if (n / divisor == 1) {
            return n;
        }
        return (n % divisor > 0) ? 0 : reverseFactorial(n / divisor, divisor + 1);
    }

    public static String formatFactorial(int n) {
        return (n == 0) ? "None" : String.format("%d!", n);
    }

}
